bom/build.gradle

```diff
@@ -0,0 +1,16 @@
+plugins {
+    id 'io.deephaven.project.register'
+}
+
+description = 'Deephaven Bill of Materials'
+
+dependencies {
+    constraints { constraint ->
+        project.rootProject
+            .subprojects
+            .findAll {  p -> io.deephaven.project.ProjectType.isPublic(p) }
+            .each { p ->
+                constraint.api p
```

Comment:
if there's any runtime-only dependencies, they should get runtime constraints too.

Reply:
Good ticket to track - https://github.com/deephaven/deephaven-core/issues/2050